src/components/app/ProfileSharing.js

```diff
@@ -2,6 +2,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+// @flow
+
```

Comment:
1) I retyped `PropTypes` using flow typings where ever I could find them.  
2) However, when I attempted to remove the PropTypes declaration for `ProfileSharingCompositeButton` I got an eslint error of the form
```
   75:13  error  'dataSource' is missing in props validation          react/prop-types
   75:25  error  'hash' is missing in props validation                react/prop-types
  137:13  error  'profile' is missing in props validation             react/prop-types
  137:22  error  'predictURL' is missing in props validation          react/prop-types
  171:19  error  'onProfilePublished' is missing in props validation  react/prop-types
```
Not sure how to resolve this error.
3) Still need to do
4) Done